lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -4,153 +4,621 @@
+  /// to the event to be retained by user code.
+  ///
+  /// For example, `setState` callbacks are fired after a component updates as a result of the
+  /// new state being passed in - and since component updates are not guaranteed to by synchronous, any
+  /// reference to a `SyntheticEvent` within that callback could have been recycled by ReactJS internals.
+  ///
+  /// You can use `persist()` to ensure access to the event properties within the callback as shown in
+  /// the second example below.
+  ///
+  /// __Without persist()__
+  /// ```dart
+  /// void handleClick(SyntheticMouseEvent event) {
+  ///   print(event.type); // => "click"
+  ///
+  ///   setState({}, () {
+  ///     print(event.type); // => null
+  ///     print(event.isPersistent); => false
+  ///   });
+  /// }
+  /// ```
+  ///
+  /// __With persist()__
+  /// ```dart
+  /// void handleClick(SyntheticMouseEvent event) {
+  ///   print(event.type); // => "click"
+  ///   event.persist();
+  ///
+  ///   setState({}, () {
+  ///     print(event.type); // => "click"
+  ///     print(event.isPersistent); => true
+  ///   });
+  /// }
+  /// ```
+  ///
+  /// See: <https://reactjs.org/docs/events.html#event-pooling>
+  @Deprecated('The modern event system does not use pooling. This does nothing.')
   external void persist();
+
+  /// Whether the event instance has been removed from the ReactJS event pool.
+  ///
+  /// > See: [persist]
+  @Deprecated('The modern event system does not use pooling. This always returns true.')
+  external bool isPersistent();
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [ClipboardEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData>
 @JS()
-@anonymous
 class SyntheticClipboardEvent extends SyntheticEvent {
+  /// [SyntheticClipboardEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticClipboardEvent` instead.
+  external factory SyntheticClipboardEvent._();
+
+  /// Holds a [DataTransfer] object, which can be used:
+  ///
+  /// - to specify what data should be put into the clipboard from the cut and copy event handlers, typically with a
+  /// setData(format, data) call;
+  /// - to obtain the data to be pasted from the paste event handler, typically with a getData(format) call.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData>
   external get clipboardData;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [KeyboardEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent>
 @JS()
-@anonymous
 class SyntheticKeyboardEvent extends SyntheticEvent {
+  /// [SyntheticKeyboardEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticKeyboardEvent` instead.
+  external factory SyntheticKeyboardEvent._();
+
+  /// Returns `true` if the `Alt` (`Option` or `⌥` on OS X) key was active when the key event was generated.
```

Comment:
#nit Getters should be documented with noun phrases ("returns" should be used only for methods), and prefer using "this" to reference the current instance

- https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-variable-getter-or-setter-comments-with-noun-phrases
- https://dart.dev/guides/language/effective-dart/documentation#prefer-using-this-instead-of-the-to-refer-to-a-members-instance

```suggestion
  /// Whether the `Alt` (`Option` or `⌥` on OS X) key was active when this event was generated.
```

Reply:
:noice: I had originally just kept the original MDN web API doc comments as it seemed to make sense to just keep it matching with what was already out there, but those tweaks were easy. Addressed in 2bc5ffa (plus a couple in 215d9ec).